fearless/ApplicationLayer/Services/WalletBalanceSubscription/WalletBalanceSubscriptionAdapter.swift

```diff
@@ -306,9 +309,6 @@ extension WalletBalanceSubscriptionAdapter: AccountInfoSubscriptionAdapterHandle
         case let .success(accountInfo):
 
             accountInfos[chainAsset.uniqueKey(accountId: accountId)] = accountInfo
-            guard chainAssets.count == accountInfos.keys.count else {
```

Comment:
according to this, you also need to clear accountInfos map every time you re-fetch data inside this class.